src/session/session_api.c

```diff
@@ -1413,13 +1413,13 @@ __wt_session_range_truncate(
      * table. If we fail to move forward or backward in a range, there are no keys in the range. In
      * either of those cases, we're done.
      */
-    if (start != NULL)
+    if (start != NULL && !F_ISSET(start, WT_CURSTD_KEY_INT))
```

Comment:
Is this check to avoid performing the cursor search as the history store already pointed to the key? Add a comment to explain it would be good.